src/components/UserButton.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Avatar,
+  Button,
+  Menu,
+  Dropdown,
+} from 'antd';
+
+import { useDispatch } from 'react-redux';
+import { Auth, Hub } from 'aws-amplify';
+import messages from './notification/messages';
+import pushNotificationMessage from '../redux/actions/notifications';
+
+const UserButton = () => {
+  const dispatch = useDispatch();
+  const [user, setUser] = useState();
+
+  const getUser = () => Auth.currentAuthenticatedUser()
+    .then((userData) => userData)
+    .catch((e) => console.log('error during getuser', e));
+
+  useEffect(() => {
+    Hub.listen('auth', ({ payload: { event } }) => {
+      switch (event) {
+        case 'signIn':
+        case 'cognitoHostedUI':
+          getUser().then((userData) => setUser(userData));
+          break;
+        case 'signOut':
+          setUser(null);
+          break;
+        case 'signIn_failure':
+        case 'cognitoHostedUI_failure':
+          dispatch(pushNotificationMessage('error', messages.signInError, 5));
+          break;
+        default:
+          break;
+      }
+    });
+
+    getUser().then((userData) => setUser(userData));
+  }, []);
+
+  const content = () => (
+    // <button onClick={() => Auth.federatedSignIn()}>Open Hosted UI</button>
+    <Menu>
+      <Menu.ItemGroup key='g1' title={`Signed in as ${user.attributes.name}`} />
+      <Menu.Item key='profile' disabled>Your profile</Menu.Item>
+      <Menu.Item key='settings' disabled>Settings</Menu.Item>
+      <Menu.Divider />
+      <Menu.Item key='logout' onClick={async () => Auth.signOut()}>Sign out</Menu.Item>
+    </Menu>
+  );
+
+  Hub.listen('auth', ({ payload: { event } }) => {
+    switch (event) {
+      case 'signIn':
+      case 'cognitoHostedUI':
+        getUser().then((userData) => setUser(userData));
+        break;
+      case 'signOut':
+        setUser(null);
+        break;
+      case 'signIn_failure':
+      case 'cognitoHostedUI_failure':
```

Comment:
what is the difference between `'signIn_failure'` and `'cognitoHostedUI_failure'`?

Reply:
`signIn` and `signIn_failure` are only used when we use a login page that is hosted by us on the UI and ask Cognito to authenticate. The corresponding `cognitoHostedUI` and `cognitoHostedUI_failure` events are used currently when we redirect users to a sign in portal.